src/ontoweaver/make_labels.py

```diff
@@ -0,0 +1,176 @@
+import logging
+import abc
+import re
+
+from . import errormanager, exceptions
+
+logger = logging.getLogger("ontoweaver")
+
+class ReturnCreate:
+    """Class for returning the extracted cell value of the transformer, along with the edge type, target node type,
+    and properties of the target node type.
+    """
+
+    def __init__(self, extracted_cell_value = None, edge_type = None, target_node_type = None, properties_of = None):
+        """
+        Initializes the ReturnCreate object.
+
+        Args:
+            extracted_cell_value: The extracted cell value of the transformer. Can serve as ID of the node, ID of the edge, or value of property.
+            edge_type: The type of the edge.
+            target_node_type: The type of the target node.
+            properties_of: The properties of the returned target node type.
+        """
+
+        self.extracted_cell_value = extracted_cell_value # Can serve as ID of the node, ID of the edge, or value of property.
+        self.edge_type = edge_type
+        self.target_node_type = target_node_type
+        self.target_element_properties = properties_of
+
+class LabelMaker(errormanager.ErrorManager, metaclass=abc.ABCMeta):
+    """Interface for defining the correct labels (types) for each extracted value.
+    The class is responsible for validating the extracted cell value and returning the appropriate label.
+    """
+
+    def __init__(self, raise_errors: bool = True):
+        """
+        Initializes the LabelMaker object.
+
+        Args:
+            raise_errors (bool): if True, will raise an exception when an error is encountered, else, will log the error and try to proceed.
+        """
+        super().__init__(raise_errors)
+
+    def check_attributes(self, **kwargs):
+        """
+        Check if all attributes expected in the __init__ are set. If not, raise an error.
+        """
+
+        # Get attributes that are already in self.__dict__ and are None
+        expected_keys = {k for k, v in self.__dict__.items() if not k.startswith('__') and v is None}
+
+        # Update only expected keys if they are present in kwargs
+        for k in expected_keys:
+            if k in kwargs:
+                setattr(self, k, kwargs[k])
+
+        # Find attributes that are still None because they were expected but not provided in kwargs
+        missing_keys = [k for k in expected_keys if getattr(self, k) is None]
+
+        # Raise an error for any missing expected attributes
+        if missing_keys:
+            for k in missing_keys:
+                self.error(
+                    f"Attribute {k} not set.",
+                    section=f"{self.__class__.__name__}.check",
+                    exception=exceptions.TransformerDataError
+                )
+
+    @abc.abstractmethod
+    def call(self, validate, returned_value, multi_type_dict, branching_properties = None, **kwargs):
```

Comment:
`kwargs` may not be necessary here, as the call is responsible for handling the data, and the constructor for handling the parameters.